### PR TITLE
Introduce pre-commit to the codebase

### PR DESCRIPTION
**Pull request by glasserc**
Experiment to introduce pre-commit to try to address #419.

To use:

- ``pip install --user pre-commit``
- ``pre-commit install`
- ``rm .git/hooks/pre-commit.legacy``
- Edit some file -- try tweaking spacing to trigger prettier, or introducing an unused variable to annoy eslint
- Stage the file (or only stage part of it -- only staged parts will be linted)
- Try to commit, or ``pre-commit run``, or ``pre-commit run eslint``/``pre-commit run prettier``

### CODE REVIEW

**Comment by glasserc** [comment]
Although pre-commit allows/encourages the use of upstream repos as lint "hooks", we don't take advantage of that in this PR, because eslint doesn't expose the right metadata, and prettier hasn't exposed it yet (although will in the next release). If those two things change, we can stop npm installing these tools manually using packages.json and start installing them automatically using pre-commit. That approach would have some trade-offs; we'd have to update our CI infra to run the commands the same way, and greenkeeper would no longer bug us about new versions of prettier and eslint. For now I'm changing as little as possible for the sake of this PR.

File: README.md [hunk 1]
```diff
@@ -31,6 +31,11 @@ Kinto-based systems.
 
 NodeJS v4+ and npm 2.14+ should be installed and available on your machine.
 
+Various pre-commit hooks are available to help prevent you from
+pushing sub-optimal code; to use these, ``pip install --user
```
Comment (n1k0) [inline README.md]: Hmm requiring python seems slightly bold here. No equivalent npm package available?

Comment (glasserc) [inline README.md]: Well, the tool is written in Python, so one way or another, you'll have to install Python. It seems that pre-commit can be installed itself using `brew`, but not everyone uses OS X, and I guess there's a `curl` installation option as well. I didn't worry about it too much because this is a tool for administering Kinto, which is itself written in Python, so I think there's a good chance you'll have Python and even `pip` available on your machine.

Comment (glasserc) [inline README.md]: N.B. There are a couple npm packages called `pre-commit` but they appear to be unrelated to the tool by Yelp.

**Comment by glasserc** [comment]
N.B. without installing the `pre-commit` tool, you can still commit as normal, just without pre-commit hooks. This isn't really an "attack vector" because you can always `git commit -n` to disable existing hooks.

File: README.md (hunk 1 above)
Comment (n1k0) [inline README.md]: Sure high are chances people have a working python environment. Ship it!
